### PR TITLE
Filter Video view by date

### PR DESCRIPTION
**Pull request by thwrk**
## Purpose

The user need to filter the video views by a given date range. 

## Proposal

We add a date picker and we query the backend to give us ony the video views within the given date range. 

- [X] Add a date picker
- [X]  Query the backend with the date range from the date picker
- [X] Add a new ui component showing the total amount of view given a time range and a list of videos

### CODE REVIEW

File: src/frontend/apps/web/pages/index.tsx [hunk 1]
```diff
@@ -27,11 +30,28 @@ const Web: NextPageWithLayout = () => {
     "uuid://dd38149d-956a-483d-8975-c1506de1e1a9",
     "uuid://e151ee65-7a72-478c-ac57-8a02f19e748b",
   ];
+  const title: String = "filter dates:"
+
+  const [since, setSince] = useState(0)
+  const [until, setUntil] = useState(9999999999999)
```
Comment (wilbrdt) [inline src/frontend/apps/web/pages/index.tsx:36]: Couldn't we use something else to initialize `since` and `until` ? Like the first and last day of the dataset?

File: src/frontend/packages/ui/DateContext.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { createContext } from 'react';
+
+type DateContext = {
+    since: number,
+    until: number,
+}
+export const DateContext = createContext({since:0, until:9999999999999});
```
Comment (wilbrdt) [inline src/frontend/packages/ui/DateContext.ts:7]: Same remark for initialization, and it's missing a newline at the end of file 
```suggestion
export const DateContext = createContext({since:0, until:9999999999999});

```

File: src/frontend/packages/ui/filter/DateRangePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    if (value) {
+      onDateChange(value[0].getTime(), value[1].getTime())
+    }
+  }
+return (
+     <>
+        <p>{title}</p>
+        <RSuiteDateRangePicker
+          format="yyyy-MM-dd HH:mm:ss"
+          defaultCalendarValue={[new Date('2023-03-01 00:00:00'), new Date('2022-05-01 23:59:59')]}
```
Comment (wilbrdt) [inline src/frontend/packages/ui/filter/DateRangePicker.tsx:25]:
```suggestion
          defaultCalendarValue={[new Date('2023-03-01 00:00:00'), new Date('2023-05-01 23:59:59')]}
```

File: src/frontend/packages/ui/filter/DateRangePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          onChange={onDateValueChange}
+
```
Comment (wilbrdt) [inline src/frontend/packages/ui/filter/DateRangePicker.tsx:27]: We could probably add `disableFuture` validation

File: src/frontend/packages/ui/video/Total.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                      new Date(firstVideoDailyViews[0].day).getTime() 
+                      : 0
+      const until = firstVideoDailyViews.length > 0 ? 
+                      new Date(firstVideoDailyViews[firstVideoDailyViews.length -1].day).getTime() 
+                      : 0
+      return {since, until}
+    } else {
+      return {since: 0, until: 0}
+    }
+  }
```
[the remark this answers is not in the document]
Comment (thwrk) [reply src/frontend/packages/ui/video/Total.tsx]: Just to let know the why of the previous implementation:
> If the user filter videos over the last 7 days and only three video where made in the last 7 days, the components would show the number of view in total "over 3 days" ( total of days where videos where made ) now it will show "over 7 days" ( the number of days of the selected periode)

( from https://github.com/openfun/warren/commit/5c50d68ccffe269d1d886c6c16cbc568d16f83c0 comment )

I guess it doesn't matter since we will only show totals numbers

File: src/backend/warren/api/v1/video/__init__.py [hunk 6]
```diff
@@ -30,7 +30,7 @@ class VideoViews(BaseModel):
 
 
 @router.get("/{video_id:path}/views")
-async def views(video_id: IRI) -> VideoViews:
+async def views(video_id: IRI, since: int = 0, until: int = 9999999999999) -> VideoViews:
     """Video views."""
     query_params = {
         "query": {
```
[the remark this answers is not in the document]
Comment (thwrk) [reply src/backend/warren/api/v1/video/__init__.py:36]: Fixed : https://github.com/openfun/warren/commit/a7512dafc0c9a5e11a9eeba96cd9370e1429ab0e

File: src/frontend/packages/ui/DateContext.ts (hunk 2 above)
Comment (thwrk) [reply src/frontend/packages/ui/DateContext.ts]: Also default to last week now

File: src/frontend/packages/ui/video/Total.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    return <span>Loading...</span>;
+
+  return (
+    <>
+      <h1>Total Views</h1>
+      <h2>{calculateTotalViewOnDateRange(videoViewStore, videoIds, since, until)} Views</h2>
+      <h2>for {videoIds.length} Videos</h2>
+      <h2>over {daysBetween(getDateRange(videoViewStore))} days</h2>
+      
+    </>
```
[the remark this answers is not in the document]
Comment (thwrk) [reply src/frontend/packages/ui/video/Total.tsx:83]: Fixed with https://github.com/openfun/warren/commit/3934947d8178864a469399a4514d765feb4ac13f

File: src/backend/warren/api/v1/video/__init__.py [hunk 8]
```diff
@@ -30,7 +30,7 @@ class VideoViews(BaseModel):
 
 
 @router.get("/{video_id:path}/views")
-async def views(video_id: IRI) -> VideoViews:
+async def views(video_id: IRI, since: int = 0, until: int = 9999999999999) -> VideoViews:
```
[the remark this answers is not in the document]
Comment (thwrk) [reply src/backend/warren/api/v1/video/__init__.py]: Fixed with b7a9cf3

File: src/frontend/packages/ui/video/Views.tsx [hunk 9]
```diff
@@ -17,11 +20,11 @@ type VideoViewsResponse = {
   daily_views: Array<DailyViewsResponseItem>;
 };
 
-type DailyViewsProps = {
+type DailyViewsAreaGraphProps = {
   videoIds: Array<string>;
 };
 
-export const DailyViews = ({ videoIds }: DailyViewsProps) => {
+export const DailyViewsAreaGraph = ({ videoIds }: DailyViewsAreaGraphProps) => {
```
Comment (jmaupetit) [inline src/frontend/packages/ui/video/Views.tsx:27]: Why did you change the component name?

File: src/frontend/apps/web/tsconfig.json [hunk 10]
```diff
@@ -1,5 +1,5 @@
 {
   "extends": "tsconfig/nextjs.json",
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "../../packages/ui/DateContext.ts"],
```
[the remark this answers is not in the document]
Comment (thwrk) [reply src/frontend/apps/web/tsconfig.json:3]: I think that's my IDE trying to be smart and me trusting it blindly, can be removed

**Comment by jmaupetit** [comment]
This work has been revamped in #9
